src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,189 @@
+- Introduction of new options in the integration hooks.
+- Introduction of new functionalities in `astro/app`, notably used for creating new adapters.
+
+:::note
+Minor releases are meant to include only non-breaking changes. However, *in rare cases* Astro will include an upgrade of the minimum required Node.js version in a minor release (e.g. from `v18.4.0` to `v18.7.0`).
+
+Astro is committed to adhering to semantic versioning and will include an upgrade to a new version of Node.js in a minor release **only after careful analysis of a variety of factors** including, but not limited to, the number of Astro projects running on incompatible versions, the security risk to developers of supporting an older version of Node.js, and the impact on Astro's core functionalities.
+:::
+
+A minor release may also include smaller, patch changes at the same time.
+
+### Major changes
+
+Major releases will include breaking changes to at least some existing code. These breaking changes are always documented in an ["Upgrade to vX" guide](#upgrade-guides) in Astro.
+
+Major releases allow Astro to make significant changes not only to internal logic, but also to intended behavior and usage. Documentation will be updated to reflect the latest version only, and **static, unmaintained snapshots of older docs** are available as a historical record for older projects that are not yet upgraded.
+
+Major releases include changes such as:
+
+- Upgrade of minimum required Node.js version (minor or major), e.g. from `v18.4.0` to `v20.0.0`.
+- Removal of previously deprecated functionalities.
+- Changes of existing functionalities.
+- Changes of existing options in the integration hooks.
+- Changes of existing options and functionalities in `astro/app`, notably used for creating new adapters.
+
+A major release may also include some non-breaking changes and improvements that would normally be released separately in a minor or patch release.
+
+### Exceptions
+
+- **Experimental features**. Releasing versions of Astro without adhering to semantic versioning allows Astro developers the greatest flexibility to explore, and even radically change course, during the development of experimental features. Therefore, the behavior of these features can break in minor and patch changes.
+
+  These features are usually accompanied by ongoing, public [Request for Consideration (RFC) stage 3](https://github.com/withastro/roadmap#stage-3-rfc--development). It is expected that beta users will follow for updates, and leave early feedback to help guide development of these features.
+
+  Once these features are out of their experimentation period, they will follow the normal semantic versioning contract.
+
+- **Improvements to the documentation** (e.g. reference and error messages). They are built from source from the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.
```

Comment:
```suggestion
- **Improvements to the documentation** (e.g. reference and error messages). They are built from source for the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.
```
I think this should be *for* the docs repository, because they're not built *from* the docs repository. They're pulled in.